internal/controller/linodevpc_controller.go

```diff
@@ -171,7 +171,8 @@ func (r *LinodeVPCReconciler) reconcile(
 		logger = logger.WithValues("vpcID", *vpcScope.LinodeVPC.Spec.VPCID)
 
 		err = r.reconcileUpdate(ctx, logger, vpcScope)
-		if err != nil && !reconciler.HasConditionSeverity(vpcScope.LinodeVPC, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+		if err != nil && !reconciler.HasStaleCondition(vpcScope.LinodeVPC, clusterv1.ReadyCondition,
+			reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultVPCControllerReconcileTimeout)) {
```

Comment:
I think we should just use `reconciler.DefaultVPCControllerReconcileTimeout` and remove `reconciler.DefaultTimeout()` as I don't think it will be useful anytime. I don't think we ever want to set global default timeout and use it.

Reply:
Do we want to do that in all of the controllers?

Reply:
Yeah, we can do this for all the controllers (places where you have changed in this PR) to keep things simple.

Reply:
I just realized that removing reconciler.DefaultTimeout() will break a bunch of test cases. For some of cases, we are manually setting the r.ReconcileTimeout to 1 nano sec to trigger an immediate timeout and see how that gets handled. This is done is all of the controllers test cases. 

One example: https://github.com/linode/cluster-api-provider-linode/blob/7f99e91f3eae2fcf79455a9c567b094eeef7c253/internal/controller/linodefirewall_controller_test.go#L142

Reply:
Ah ok, that might be why its added in first place, hmm